source/server/configuration_impl.h

```diff
@@ -111,13 +131,19 @@ class MainImpl : Logger::Loggable<Logger::Id::config>, public Main {
 
   void initializeStatsSinks(const envoy::config::bootstrap::v3::Bootstrap& bootstrap,
                             Instance& server);
+  /**
+   * Initialize watchdog(s). Call before accessing any watchdog configuration.
+   */
   void initializeWatchdogs(const envoy::config::bootstrap::v3::Bootstrap& bootstrap,
                            Instance& server);
 
   std::unique_ptr<Upstream::ClusterManager> cluster_manager_;
   std::list<Stats::SinkPtr> stats_sinks_;
   std::chrono::milliseconds stats_flush_interval_;
   std::unique_ptr<Watchdog> watchdog_;
+  std::unique_ptr<Watchdog> aux_watchdog_;
+  std::unique_ptr<Watchdog> worker_watchdog_;
+  bool multi_watchdog_;
```

Comment:
Since the values of these fields are so tied together, consider using a type that enforces the invariants you want. Here, for instance, multi_watchdog_ tells you which of `watchdog_` and (`aux_watchdog_`, `worker_watchdog_`) is valid. Consider using `absl::variant` as a type-safe union that will subsume these (and save some space!).

Reply:
I thought about this for a while... 

Using an `absl::variant` but it seems like they're be a good amount of boilerplate since I'd have to wrap the `aux_watchdog_` and `worker_watchdog_` together somehow. 

I could alternatively jump to using two watchdogs systems (vs the single watchdog system), while still supporting  the single `watchdog` field. It'd be much easier to "emulate" most of the functionality of the single watchdog system using the two watchdog system.

This would make things a lot cleaner, but would break any cross-watchdog policies (such as `multikill` triggering under a combination of `main_thread_watchdog` and `worker_watchdog`), but that functionality doesn't seem too helpful.

I'm leaning towards the latter.

Reply:
I've made the system always have two guard dogs which removes this issue. 

If the user specifies the `watchdog` field (the old system), then we'll just have the two guarddogs have the same configuration.